ocrd_detectron2/segment.py

```diff
@@ -187,45 +187,47 @@ class id to a new PAGE region type (and subtype).
-                if diff > 0:
-                    page_image_raw = crop_image(
-                        page_image_raw,
-                        (0, int(np.floor(diff / 2)),
-                         page_image_raw.width,
-                         page_image_raw.height - int(np.ceil(diff / 2))))
-                else:
-                    page_image_bin = crop_image(
-                        page_image_bin,
-                        (0, int(np.floor(-diff / 2)),
-                         page_image_bin.width,
-                         page_image_bin.height - int(np.ceil(-diff / 2))))
+            # check wether input image is binarized
+            if page_image_info.photometricInterpretation == "1":
+                # for morphological post-processing, we will need the binarized image, too
+                page_image_bin, _, _ = self.workspace.image_from_page(
+                    page, page_id,
+                    feature_selector='binarized')
+                # workaround for OCR-D/core#687:
+                if 0 < abs(page_image_raw.width - page_image_bin.width) <= 2:
+                    diff = page_image_raw.width - page_image_bin.width
+                    if diff > 0:
+                        page_image_raw = crop_image(
+                            page_image_raw,
+                            (int(np.floor(diff / 2)), 0,
+                            page_image_raw.width - int(np.ceil(diff / 2)),
+                            page_image_raw.height))
+                    else:
+                        page_image_bin = crop_image(
+                            page_image_bin,
+                            (int(np.floor(-diff / 2)), 0,
+                            page_image_bin.width - int(np.ceil(-diff / 2)),
+                            page_image_bin.height))
+                if 0 < abs(page_image_raw.height - page_image_bin.height) <= 2:
+                    diff = page_image_raw.height - page_image_bin.height
+                    if diff > 0:
+                        page_image_raw = crop_image(
+                            page_image_raw,
+                            (0, int(np.floor(diff / 2)),
+                            page_image_raw.width,
+                            page_image_raw.height - int(np.ceil(diff / 2))))
+                    else:
+                        page_image_bin = crop_image(
+                            page_image_bin,
+                            (0, int(np.floor(-diff / 2)),
+                            page_image_bin.width,
+                            page_image_bin.height - int(np.ceil(-diff / 2))))
 
             # ensure RGB (if raw was merely grayscale)
             if page_image_raw.mode == '1':
                 page_image_raw = page_image_raw.convert('L')
             page_image_raw = page_image_raw.convert(mode='RGB')
-            page_image_bin = page_image_bin.convert(mode='1')
+            page_image_bin = page_image_raw.convert(mode='1')
```

Comment:
That's not going to work well: binarization is usually more than a simple conversion – you have to find the best threshold, ideally localized across the image.